src/displayapp/screens/WatchFaceDigital.cpp

```diff
@@ -153,4 +169,28 @@ void WatchFaceDigital::Refresh() {
     lv_obj_realign(stepValue);
     lv_obj_realign(stepIcon);
   }
+
+  currentWeather = weatherService.Current();
+
+  if (currentWeather.IsUpdated()) {
+    auto optCurrentWeather = currentWeather.Get();
+    if (optCurrentWeather) {
+      int16_t temp = optCurrentWeather->temperature;
+      char tempUnit = 'C';
+      if (settingsController.GetWeatherFormat() == Controllers::Settings::WeatherFormat::Imperial) {
+        temp = Controllers::SimpleWeatherService::CelsiusToFahrenheit(temp);
+        tempUnit = 'F';
+      }
+      temp = temp / 100 + (temp % 100 >= 50 ? 1 : 0);
+      lv_label_set_text_fmt(temperature, "%d°%c", temp, tempUnit);
+      lv_label_set_text(weatherIcon, Symbols::GetSymbol(optCurrentWeather->iconId));
+      lv_obj_realign(temperature);
+      lv_obj_realign(weatherIcon);
+    }
+  } else {
```

Comment:
Is the else branch one too late. For me it reads: if the current weather isn't updated then set it to empty string. I think this would only be needed of the optional has no value

Reply:
`weatherService.Current()` returns a reference to the current weather instance only if the weather is "recent enough" (quick look seems to show 24 hours). Otherwise it returns an empty value.

In either case it considers it "updated", so I think you're right. The only case when the `else` will get called is if weather data has _never_ been sent since the init of the service.

Therefore it makes sense to move the `else` clause inside the `IsUpdated` conditional check and clear the values when `currentWeather` is empty. I'll test this out and push the change.

Reply:
It seems to work, so I've pushed the change. I also added a new commit to fix the PineTimeStyle face (which has the same issue).

Reply:
Fixed by #1965.